### PR TITLE
Pooling layer docs

### PR DESCRIPTION
**Pull request by benanne**
Cleaned up the docs for the pooling layers a bit further (@ebenolson already did a bunch of work on them before). This is as preparation for the corrmm/cuda_convnet/dnn docs, which I will probably do next.

### CODE REVIEW

**Comment by benanne** [comment]
Anyone wanna have a quick look at this? Anyone who wants to, it's just useful to have a second set of eyes looking at it.  It would be appreciated! :)

File: lasagne/layers/pool.py [hunk 1]
```diff
@@ -72,39 +72,45 @@ def pool_output_length(input_length, pool_size, stride,
...
+    incoming : a :class:`Layer` instance or tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 3D tensor.
+
+    pool_size : integer or iterable
+        The length of the pooling region
+
+    stride : integer, iterable or None
+        The stride between sucessive pooling regions.
+        If `None`, stride = pool_size.
```
Comment (ebenolson) [inline lasagne/layers/pool.py]: Is there a standard for when to use code quotes? Why here but not 2 lines above?

File: lasagne/layers/pool.py [hunk 2]
```diff
@@ -326,9 +363,29 @@ def get_output_for(self, input, **kwargs):
...
+        (i.e. mean-pooling) and can be replaced by any other aggregation
+        function.
+
+    **kwargs
+        Any additional keyword arguments are passed to the :class:`Layer`
+        superclass.
+
+    Notes
+    -----
+    This layer requires that the size of the axis along which it groups units
```
Comment (ebenolson) [inline lasagne/layers/pool.py]: Copypaste artifact

File: lasagne/layers/pool.py [hunk 3]
```diff
@@ -217,28 +229,44 @@ def get_output_for(self, input, **kwargs):
...
+    :class:`Conv1DLayer` and :class:`Conv2DLayer`. The layer can be used to
+    implement maxout.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or tuple
+        The layer feeding into this layer, or the expected input shape.
+
+    pool_size : integer
+        the size of the pooling windows, i.e. the number of features / feature
```
Comment (ebenolson) [inline lasagne/layers/pool.py]: we refer to them as pooling "regions", "windows", and "groups" in various places, probably should choose one.

**Comment by ebenolson** [comment]
Made a couple minor comments, looks pretty good to me (I didn't try compiling it though, just looked through the diff).

File: lasagne/layers/pool.py (hunk 1 above)
Comment (benanne) [inline lasagne/layers/pool.py]: I'm not sure to be honest. I think you're even supposed to use double backticks (that's what you're talking about right?). But I think the rules for the actual type specification may be different. Maybe someone with more numpydoc experience can shed some light on this.

File: lasagne/layers/pool.py (hunk 2 above)
Comment (benanne) [inline lasagne/layers/pool.py]: Which line exactly? I think I'm reading over it.

File: lasagne/layers/pool.py (hunk 3 above)
Comment (benanne) [inline lasagne/layers/pool.py]: Good point. What name is most common in literature? I guess I'm fine standardizing on either "regions" or "windows" (I think "groups" is a bit of a vague term).

The term 'window' is also polyvalent to some extent, so I guess 'region' would be the least confusing. I'll go with that, unless anyone wants to have another name discussion of course ;)

**Comment by benanne** [comment]
Thanks!

File: lasagne/layers/pool.py (hunk 2 above)
Comment (ebenolson) [inline lasagne/layers/pool.py]: The whole notes section, `GlobalPoolLayer` doesn't have a pool size

Comment (benanne) [inline lasagne/layers/pool.py]: Whoops :) Thanks for catching this!

**Comment by benanne** [comment]
I made the proposed changes. Since this is sort of small I'll merge it tomorrow provided that there are no further comments and Travis passes.

File: lasagne/layers/pool.py (hunk 3 above)
Comment (f0k) [inline lasagne/layers/pool.py]: Did I hear "name discussion"? :)

"groups" are used in cuda-convnet for grouped convolution, where you learn kernels for subsets of the input channels. In this case, you don't specify the size of the pooling region/window/thing, but the number of groups, and if you use the same number for each convolutional layer, you get a multi-column architecture.
So this notion of "groups" is closely related to what we're doing here if axis=1, but as it doesn't have to be axis=1 and if we want to have a common term, I think "region" is a good choice. "window" is something I would associate with the last axis/axes in bc01* notation.

**Pull request merged**